### PR TITLE
feat: add message batcher for terminal process

### PR DESCRIPTION
**Pull request by zhaomenghuan**
#### What it does

Batch the output of the terminal, combine the output content within 16ms and send it to the front end in batches, so as to avoid blocking RPC communication when the terminal executes a lot of output commands. Inspired by [opensumi#304](https://github.com/opensumi/core/pull/304) and [hyper 3](https://hyper.is/blog).

#### How to test

- Click the new terminal menu to create a new terminal
- Enter the `find ~` command, then abort the command, the terminal stops immediately

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: zhaomenghuan <1028317108@qq.com>

### CODE REVIEW

**Review by vince-fugnitto — COMMENTED** [review]
@zhaomenghuan it seems like the macos tests are failing with this change.

**Comment by zhaomenghuan** [comment]
> @zhaomenghuan it seems like the macos tests are failing with this change.

@vince-fugnitto The code has been adjusted, and the test case error has been fixed.

File: packages/process/src/node/terminal-process.ts [hunk 1]
```diff
@@ -45,6 +45,12 @@ export enum NodePtyErrors {
     ENOENT = 'No such file or directory'
 }
 
+// Max size of a data batch. Note that this value can be exceeded by ~4k (chunk sizes seem to be 4k at the most)
+export const BATCH_MAX_SIZE = 200 * 1024;
+
+// Max duration to batch data before sending it to the renderer process.
+export const BATCH_DURATION_MS = 16;
```
Comment (msujew) [inline packages/process/src/node/terminal-process.ts:52]: While testing this, I've noticed that 16 ms is way too short of a delay. Every other action was heavily delayed, often until the command running in the terminal has ended. Changing this to `64` has improved this behavior dramatically, basically enabling normal app interaction.

**Review by msujew — COMMENTED** [review]
Looks quite good. However with the `16 ms` duration the app was still unusable (web mode). Increasing the duration improved responsiveness without noticeably impacting the terminal output.

**Review by msujew — APPROVED** [review]
Actually, I'll dismiss my previous comment, as that seems to be related to an issue I experienced in Gitpod only. I can't reproduce that anymore, so I'll approve it.

- [x] The changes allow for app interaction even when faced with a lot of terminal output
- [x] I couldn't find any regressions using the changes.

@vince-fugnitto Do you mind testing this as well? It seems like this could be influenced by the environment Theia runs in.

**Comment by paul-marechal** [comment]
I am not sold on putting those changes at the `TerminalProcess` level: Ultimately we want to control the size and timing of the RPC requests we make to the frontend. We also use independent ring buffer readers that may not benefit from this change.

May I propose an alternative change? https://github.com/eclipse-theia/theia/pull/11449

**Pull request closed without merging**